doc/examples/00_prepare_timeseries.ipynb

```diff
@@ -276,81 +276,312 @@
+    "First let's invent some data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "80b04dd3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# monthly discharge volumes\n",
+    "# volume on 1 february 2022 at 00:00 is the volume for month of january\n",
+    "index = pd.date_range(\"2022-02-01\", \"2023-01-01\", freq=\"MS\")\n",
+    "discharge = np.arange(12) * 100\n",
     "\n",
+    "Q0 = pd.Series(index=index, data=discharge)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "5a1c1a8e",
+   "metadata": {},
+   "source": [
+    "Next, use `timestep_weighted_resample` to calculate a daily time series for\n",
+    "well pumping.\n",
     "\n",
-    "_**Note:** in terms of performance the pandas methods are undoubtedly faster._"
+    "**Note**: _the unit of the resulting daily time series has not changed! This\n",
+    "means the daily value is assumed to be the same as the monthly value. So take\n",
+    "note of your units when modeling resampled time series with Pastas!_"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "e8c5d811",
+   "id": "c99e071e",
    "metadata": {},
    "outputs": [],
    "source": [
-    "def pandas_sample(series, freq):\n",
-    "    series = series.copy()\n",
-    "    t_offset = ps.utils._get_time_offset(series.index, freq).value_counts().idxmax()\n",
-    "    new_idx = pd.date_range(\n",
-    "        series.index[0].floor(freq) + t_offset,\n",
-    "        series.index[-1].floor(freq) + t_offset,\n",
-    "        freq=freq,\n",
-    "    )\n",
-    "    return series.reindex(new_idx)\n",
+    "# define a new index\n",
+    "new_index = pd.date_range(\"2022-01-01\", \"2022-12-31\", freq=\"D\")\n",
     "\n",
+    "# resample to daily data\n",
+    "Q_daily = ps.ts.timestep_weighted_resample(Q0, new_index)\n",
     "\n",
-    "def pandas_nearest(series, freq, tolerance=None):\n",
-    "    series = series.copy()\n",
-    "    # Create equidistant timeseries with Pandas\n",
-    "    idx = pd.date_range(\n",
-    "        series.index[0].floor(freq), series.index[-1].ceil(freq), freq=freq\n",
-    "    )\n",
-    "    spandas = series.reindex(idx, method=\"nearest\", tolerance=tolerance)\n",
-    "    return spandas\n",
+    "# plot the comparison\n",
+    "ax = Q0.plot(marker=\"o\", label=\"original monthly series\")\n",
+    "Q_daily.plot(ax=ax, label=\"resampled daily series\")\n",
+    "ax.legend()\n",
+    "ax.grid(True)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2d1af6e1",
+   "metadata": {},
+   "source": [
+    "### Example 2: Resample precipitation between 9AM-9AM to 12AM-12AM\n",
```

Comment:
Maybe we can add a simple calculation here to confirm that `timestep_weighted_resample()` does the right thing. So: `x[i-1] * 9/24 + x[i] * 15/24`.

Reply:
added this myself :)